sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1449,6 +1449,15 @@ object SQLConf {
     .intConf
     .checkValues((1 to 9).toSet + Deflater.DEFAULT_COMPRESSION)
     .createWithDefault(Deflater.DEFAULT_COMPRESSION)
+
+  val PARALLEL_FILE_LISTING_IN_COMMMANDS =
+    buildConf("spark.sql.parallelFileListingInCommands.enabled")
```

Comment:
Currently, in the master, this listing happens only in the ANALYZE SQL command, so this name fit the case now. But, in a future, if we probably add a new interface for the listing (e.g., `SparkSession.analyzeTable`, `SparkSession.analyzeColumn`, ...), this name a little confuses users? So, how about `spark.sql.parallelFileListingInStatsComputation.enabledl`? @gatorsmile

Reply:
I can make this change..